numpy/core/_type_aliases.py

```diff
@@ -1,224 +1,263 @@
-                  ('csingle', 'cfloat'),
-                  ('singlecomplex', 'cfloat'),
-                  ('float_', 'double'),
-                  ('intc', 'int'),
-                  ('uintc', 'uint'),
-                  ('int_', 'long'),
-                  ('uint', 'ulong'),
-                  ('cfloat', 'cdouble'),
-                  ('longfloat', 'longdouble'),
-                  ('clongfloat', 'clongdouble'),
-                  ('longcomplex', 'clongdouble'),
-                  ('bool_', 'bool'),
-                  ('bytes_', 'string'),
-                  ('string_', 'string'),
-                  ('str_', 'unicode'),
-                  ('unicode_', 'unicode'),
-                  ('object_', 'object')]
-    for alias, t in type_pairs:
-        allTypes[alias] = allTypes[t]
-        sctypeDict[alias] = sctypeDict[t]
-    # Remove aliases overriding python types and modules
-    to_remove = ['object', 'int', 'float',
-                 'complex', 'bool', 'string', 'datetime', 'timedelta',
-                 'bytes', 'str']
-
-    for t in to_remove:
-        try:
-            del allTypes[t]
-            del sctypeDict[t]
-        except KeyError:
-            pass
+sctypeDict = numbers_dict | symbol_dict | symbol_bytes_dict | \
+    word_dict | word_bits_dict | extra_aliases_dict
+# delete exceptions
+for s in ["O8", "O4", "S0", "U0", "V0", "datetime", "object32",
+          "object64", "string", "timedelta"]:
+    try:
+        del sctypeDict[s]
+    except KeyError:
+        pass
 
-    # Additional aliases in sctypeDict that should not be exposed as attributes
-    attrs_to_remove = ['ulong', 'long', 'unicode']
 
-    for t in attrs_to_remove:
+# Finally, we build `sctypes`` mapping
+def _build_sctypes() -> Dict[str, List[type]]:
+    def _add_array_type(typename, bits, sctypes):
         try:
-            del allTypes[t]
+            t = sctypeDict[f"{typename}{bits}"]
         except KeyError:
             pass
-_set_up_aliases()
-
-
-sctypes = {'int': [],
-           'uint': [],
-           'float': [],
-           'complex': [],
-           'others': [bool, object, bytes, str, void]}
-
-
-def _add_array_type(typename, bits):
-    try:
-        t = allTypes['%s%d' % (typename, bits)]
-    except KeyError:
-        pass
-    else:
-        sctypes[typename].append(t)
-
-def _set_array_types():
+        else:
+            sctypes[typename].append(t)
+
+    sctypes = {
+        'int': [],
+        'uint': [],
+        'float': [],
+        'complex': [],
+        'others': [bool, object, bytes, str, allTypes['void']]
+    }
+    
     ibytes = [1, 2, 4, 8, 16, 32, 64]
     fbytes = [2, 4, 8, 10, 12, 16, 32, 64]
-    for bytes in ibytes:
-        bits = 8*bytes
-        _add_array_type('int', bits)
-        _add_array_type('uint', bits)
-    for bytes in fbytes:
-        bits = 8*bytes
-        _add_array_type('float', bits)
-        _add_array_type('complex', 2*bits)
+    for b in ibytes:
+        bits = 8*b
```

Comment:
```suggestion
        bits = 8 * b
```
Although not strictly required by pep8, whitespace around the operator looks nicer.

Reply:
Sure, I agree, done!